File: AtlasWeb/Core/Modules/Tibbi_Surec/Hasta_Raporlari_Modulu/HastaNakilFormuViewModel.cs
using Infrastructure.Models;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using TTObjectClasses;

namespace Core.Models
{
    public class HastaNakilFormuViewModel
    {
        public bool? IsNew { get; set; }
        public string SubepisodeID { get; set; }
        public string ObjectID { get; set; }
        public DateTime? TalepEdildigiZaman { get; set; }
        public Guid? NakilTalepEdenKlinik { get; set; }
        public Guid? NakilEdilmekIstenilenKlinik { get; set; }
        public Guid? HastaninBulunduguKlinik { get; set; }
        public Guid? KomutaKontrolMerkezi { get; set; }
        public Guid? NakilGerceklestirmeYolu { get; set; }
        public Guid? SevkNedeni { get; set; }
        public string SevkNedeniAciklama { get; set; }
        public Guid? KabulEdenKurumIli { get; set; }
        public Guid? KabulEdenKurum { get; set; }
        public Guid? NakilKabulEdenKlinik { get; set; }
        public Guid? HastaHukumluMu { get; set; }
        public Guid? AdliVakaMi { get; set; }
        public Guid? KanGrubu { get; set; }
        public Guid? NakilTipi { get; set; }
        public Guid? DoktorIhtiyaci { get; set; }
        public Guid? BransIhtiyaci { get; set; }
        public Guid? TeyitliVakaMi { get; set; }
        public int? SistolikKanBasinci { get; set; }
        public int? DiastolikKanBasinci { get; set; }
        public Guid? Solunum { get; set; }
        public int? SolunumSayisi { get; set; }
        public string SolunumIslemi { get; set; }
        public int? GlaskowKomaSkalasi { get; set; }
        public int? Gozler { get; set; }
        public Guid Triaj { get; set; }
        public string Ates { get; set; }
        public int? NabizSayisi { get; set; }
        public Guid? Bilinc { get; set; }
        public int? KanSekeri { get; set; }
        public int? Sozel { get; set; }
        public int? Motor { get; set; }
        public object VitalBulgular { get; set; }
        public object PatolojikMuayeneBilgileri { get; set; }
        public object NakilSirasindaIstenilenMedikalIslemler { get; set; }
        public object TetkikBilgileri { get; set; }
        public object YapilanMedikalIslemler { get; set; }
        public object YapilmasiIstenilenMedikalIslemler { get; set; }
        public object NakilSirasindakiGereksinimler { get; set; }
        public Guid? TransportMalzemesi { get; set; }
        public int? MalzemeSayisi { get; set; }
        public string HastaYakiniAdi { get; set; }
        public string HastaYakiniSoyadi { get; set; }
        public string HastaYakiniTel { get; set; }
        public string HastaYakiniAdres { get; set; }
        public object EpikrizeEkAciklama { get; set; }
        public string HekimAdi { get; set; }
        public string HekimSoyadi { get; set; }
        public string HekimTel { get; set; }
        public string HekimTC { get; set; }
        public string PersonelAdi { get; set; }
        public string PersonelSoyadi { get; set; }
        public string PersonelTel { get; set; }

        public string _NabizDurumu { get; set; }
        public List<DiagnosisAndProcedureBaseModel> SevkTaniList { get; set; }
    }

    public class HastaNakilSKRSModel
    {
        public List<TTObjectClasses.SKRSKlinikler> SKRSKlinikList { get; set; }
        public List<TTObjectClasses.SKRSKurumlar> SKRSKurumList { get; set; }
        public List<TTObjectClasses.SKRSNAKILYOLU> SKRSNakilYoluList { get; set; }
        public List<TTObjectClasses.SKRSSEVKNEDENI> SKRSSevkNedeniList { get; set; }
        public List<TTObjectClasses.SKRSILKodlari> SKRSILKodlariList { get; set; }
        public List<TTObjectClasses.SKRSDurum> SKRSDurumList { get; set; }
        public List<TTObjectClasses.SKRSKanGrubu> SKRSKanGrubuList { get; set; }
        public List<TTObjectClasses.SKRSHastaNakilTipi> SKRSHastaNakilTipiList { get; set; }
        public List<TTObjectClasses.SKRSSOLUNUM> SKRSSolunumList { get; set; }
        public List<TTObjectClasses.SKRSTRANSPORTMALZEMESI> SKRSTransportMalzemesiList { get; set; }
        public List<TTObjectClasses.SKRSTRIAJKODU> SKRSTriajKoduList { get; set; }
        public List<TTObjectClasses.SKRSBilinc> SKRSBilincList { get; set; }
        public List<TTObjectClasses.SKRSPersonelBransKodu> SKRSPersonelBransList { get; set; }

        public List<DoctorInfo> DoctorList { get; set; }
    }

    public class SKRSKurumDTO
    {
        public int? KODU { get; set; }
        public Guid? ObjectID { get; set; }
        public string ADI { get; set; }
    }

    public class DoctorInfo
    {
        public Guid ObjectID { get; set; }
        public string NameSurname { get; set; }
        public string Name { get; set; }
        public string Surname { get; set; }
        public string Phone { get; set; }
        public string UniqueRefNo { get; set; }
    }
}

File: AtlasWeb/Core/Modules/Tibbi_Surec/Hasta_Raporlari_Modulu/HastaNakilFormuViewModel.ts
import { Type } from 'NebulaClient/ClassTransformer';
import { Guid } from 'NebulaClient/Mscorlib/Guid';
import { SKRSKlinikler, SKRSKurumlar, SKRSNAKILYOLU, SKRSSEVKNEDENI, SKRSILKodlari, SKRSDurum, SKRSKanGrubu, SKRSSOLUNUM, SKRSTRANSPORTMALZEMESI, EpisodeAction, DiagnosisGrid, SKRSHastaNakilTipi, SKRSTRIAJKODU, SKRSBilinc, SKRSPersonelBransKodu } from '../../../wwwroot/app/NebulaClient/Model/AtlasClientModel';
import { DiagnosisAndProcedureBaseModel } from '../Tibbi_Surec_Is_Listeleri/Yatan_Hasta_Is_Listesi/InPatientWorkListViewModel';

export class HastaNakilFormuViewModel {
    public IsNew: boolean;
    public SubepisodeID: string;
    public ObjectID: string;
    @Type(() => Date)
    public TalepEdildigiZaman: Date;
    @Type(() => Guid)
    public NakilTalepEdenKlinik: Guid;
    @Type(() => Guid)
    public NakilEdilmekIstenilenKlinik: Guid;
    @Type(() => Guid)
    public HastaninBulunduguKlinik: Guid;
    @Type(() => Guid)
    public KomutaKontrolMerkezi: Guid;
    @Type(() => Guid)
    public NakilGerceklestirmeYolu: Guid;
    @Type(() => Guid)
    public SevkNedeni: Guid;
    public SevkNedeniAciklama: string;
    @Type(() => Guid)
    public KabulEdenKurumIli: Guid;
    @Type(() => Guid)
    public KabulEdenKurum: Guid;
    @Type(() => Guid)
    public NakilKabulEdenKlinik: Guid;
    @Type(() => Guid)
    public HastaHukumluMu: Guid;
    @Type(() => Guid)
    public AdliVakaMi: Guid;
    @Type(() => Guid)
    public KanGrubu: Guid;
    @Type(() => Guid)
    public NakilTipi: Guid;
    @Type(() => Guid)
    public DoktorIhtiyaci: Guid;
    @Type(() => Guid)
    public BransIhtiyaci: Guid;
    @Type(() => Guid)
    public TeyitliVakaMi: Guid;
    public SistolikKanBasinci: number;
    public DiastolikKanBasinci: number;
    @Type(() => Guid)
    public Solunum: Guid;
    public SolunumSayisi: number;
    public SolunumIslemi: string;
    public GlaskowKomaSkalasi: number;
    public Gozler: number;
    @Type(() => Guid)
    public Triaj: Guid;
    public Ates: string;
    public NabizSayisi: number;
    @Type(() => Guid)
    public Bilinc: Guid;
    public KanSekeri: number;
    public Sozel: number;
    public Motor: number;
    @Type(() => Object)
    public VitalBulgular: Object;
    @Type(() => Object)
    public PatolojikMuayeneBilgileri: Object;
    @Type(() => Object)
    public NakilSirasindaIstenilenMedikalIslemler: Object;
    @Type(() => Object)
    public TetkikBilgileri: Object;
    @Type(() => Object)
    public YapilanMedikalIslemler: Object;
    @Type(() => Object)
    public YapilmasiIstenilenMedikalIslemler: Object;
    @Type(() => Object)
    public NakilSirasindakiGereksinimler: Object;
    @Type(() => Guid)
    public TransportMalzemesi: Guid;
    public MalzemeSayisi: number;
    public HastaYakiniAdi: string;
    public HastaYakiniSoyadi: string;
    public HastaYakiniTel: string;
    public HastaYakiniAdres: string;
    @Type(() => Object)
    public EpikrizeEkAciklama: Object;
    public HekimAdi: string;
    public HekimSoyadi: string;
    public HekimTel: string;
    public HekimTC: string;
    public PersonelAdi: string;
    public PersonelSoyadi: string;
    public PersonelTel: string;
    public _NabizDurumu: string;
    @Type(() => DiagnosisAndProcedureBaseModel)
    public SevkTaniList: Array<DiagnosisAndProcedureBaseModel> = new Array<DiagnosisAndProcedureBaseModel>();

}

export class HastaNakilSKRSModel {

    @Type(() => SKRSKlinikler)
    public SKRSKlinikList: Array<SKRSKlinikler> = new Array<SKRSKlinikler>();
    @Type(() => SKRSKurumlar)
    public SKRSKurumList: Array<SKRSKurumlar> = new Array<SKRSKurumlar>();
    @Type(() => SKRSNAKILYOLU)
    public SKRSNakilYoluList: Array<SKRSNAKILYOLU> = new Array<SKRSNAKILYOLU>();
    @Type(() => SKRSSEVKNEDENI)
    public SKRSSevkNedeniList: Array<SKRSSEVKNEDENI> = new Array<SKRSSEVKNEDENI>();
    @Type(() => SKRSILKodlari)
    public SKRSILKodlariList: Array<SKRSILKodlari> = new Array<SKRSILKodlari>();
    @Type(() => SKRSDurum)
    public SKRSDurumList: Array<SKRSDurum> = new Array<SKRSDurum>();
    @Type(() => SKRSKanGrubu)
    public SKRSKanGrubuList: Array<SKRSKanGrubu> = new Array<SKRSKanGrubu>();
    @Type(() => SKRSHastaNakilTipi)
    public SKRSHastaNakilTipiList: Array<SKRSHastaNakilTipi> = new Array<SKRSHastaNakilTipi>();
    @Type(() => SKRSSOLUNUM)
    public SKRSSolunumList: Array<SKRSSOLUNUM> = new Array<SKRSSOLUNUM>();
    @Type(() => SKRSTRANSPORTMALZEMESI)
    public SKRSTransportMalzemesiList: Array<SKRSTRANSPORTMALZEMESI> = new Array<SKRSTRANSPORTMALZEMESI>();
    @Type(() => SKRSTRIAJKODU)
    public SKRSTriajKoduList: Array<SKRSTRIAJKODU> = new Array<SKRSTRIAJKODU>();
    @Type(() => SKRSBilinc)
    public SKRSBilincList: Array<SKRSBilinc> = new Array<SKRSBilinc>();
    @Type(() => SKRSPersonelBransKodu)
    public SKRSPersonelBransList: Array<SKRSPersonelBransKodu> = new Array<SKRSPersonelBransKodu>();

    @Type(() => DoctorInfo)
    public DoctorList: Array<DoctorInfo> = new Array<DoctorInfo>();

}
export class DoctorInfo {
    @Type(() => Guid)
    public ObjectID: Guid;
    public NameSurname: string;
    public Name: string;
    public Surname: string;
    public Phone: string;
    public UniqueRefNo: string;
}